crates/solidity/inputs/language/src/definition.rs

```diff
@@ -2603,26 +2603,19 @@ codegen_language_macros::compile!(Language(
                     items = [
                         Struct(
                             name = EventDefinition,
-                            error_recovery = FieldsErrorRecovery(terminator = semicolon),
-                            fields = (
-                                event_keyword = Required(Terminal([EventKeyword])),
-                                name = Required(Terminal([Identifier])),
-                                parameters =
-                                    Optional(kind = NonTerminal(EventParametersDeclaration)),
-                                anonymous_keyword = Optional(kind = Terminal([AnonymousKeyword])),
-                                semicolon = Required(Terminal([Semicolon]))
-                            )
-                        ),
-                        Struct(
-                            name = EventParametersDeclaration,
                             error_recovery = FieldsErrorRecovery(
+                                terminator = semicolon,
                                 delimiters =
                                     FieldDelimiters(open = open_paren, close = close_paren)
                             ),
                             fields = (
+                                event_keyword = Required(Terminal([EventKeyword])),
+                                name = Required(Terminal([Identifier])),
                                 open_paren = Required(Terminal([OpenParen])),
-                                parameters = Required(NonTerminal(EventParameters)),
-                                close_paren = Required(Terminal([CloseParen]))
+                                parameters = Optional(kind = NonTerminal(EventParameters)),
```

Comment:
If `EventParameters` is `Optional`, but also has `allow_empty = true`, how would it work in this case? should it be `allow_empty = false` in this case, or make it `Required`?

note to self: add validation for this corner case later.

Reply:
My initial thinking is that these lists should be `Optional(OneOrMore)`, so that we don't end up creating empt nodes unnecissarily (for perf/memory allocation).
I wonder if we need to keep these nodes in the tree.

Happy to help fix it in v2 if you agree.

Reply:
Actually, chatting with Tony, looks like we have - to use `Required(ZeroOrMore)` everywhere, creating nodes, for us to establish binding scopes on them, even if they don't have children.

Happy to fix it as well, if you haven't already started that work.

Reply:
> Happy to fix it as well, if you haven't already started that work.

That was initially the default for https://github.com/NomicFoundation/slang/pull/650 but I changed the v2 definition to be in line with v0/v1 to limit the resulting CST diff. Happy to change it back once we start using v2 exclusively for parser codegen.

> My initial thinking is that these lists should be Optional(OneOrMore)

Yes, I changed it to be like that in #650. I probably forgot to change it in this PR. Let me do that, so that we're in line with what's happening there

Reply:
Done in https://github.com/NomicFoundation/slang/pull/653/commits/935a9ef23b459d5a95d4353a46338faad69cd44a. The rest is marked as `Optional(OneOrMore)` in #650 to retain the old behaviour but we can change it to `Required(ZeroOrMore)` once we move off DSL v1.

Reply:
Sounds good. I added this to #655